core/backend/src/TxnManager.ts

```diff
@@ -470,4 +490,14 @@ export class TxnManager {
   /** Query if there are un-saved or un-pushed local changes. */
   public get hasLocalChanges(): boolean { return this.hasUnsavedChanges || this.hasPendingTxns; }
 
+  /** Query local changes
+   * @beta
+  */
+  public queryLocalChanges(args?: QueryLocalChangesArgs): ChangeInstanceKey[] {
```

Comment:
Do you want to commit to returning a (potentially very large) array rather than returning an `Iterable<ChangeInstanceKey>` whose implementation can be changed in the future if needed?